src/cli/subcommand/split_receipt.rs

```diff
@@ -0,0 +1,477 @@
+
+use crate::cli::CommandExecute;
+
+pub(crate) const PHASE1_RECEIPT_LOCATION: &str = "/nix/uninstall-phase1.json";
+pub(crate) const PHASE2_RECEIPT_LOCATION: &str = "/nix/uninstall-phase2.json";
+
+/// Split an existing receipt into two phases, one that cleans up the Nix store (phase 2), and
+/// one that does everything else (phase 1).
+///
+/// This will produce two modified receipts -- a phase 1 receipt and a phase 2 receipt. If you
+/// run `/nix/nix-installer uninstall /nix/uninstall-phase1.json` (the default phase 1
+/// location), it will clean up everything but the Nix store and allow you to reinstall with a
+/// newer version. If you run `/nix/nix-installer uninstall /nix/uninstall-phase2.json`, then it
+/// will complete the uninstall by cleaning up the Nix store.
+#[derive(Debug, Parser)]
+pub struct SplitReceipt {
+    #[clap(
+        long,
+        env = "NIX_INSTALLER_NO_CONFIRM",
+        action(ArgAction::SetTrue),
+        default_value = "false",
+        global = true
+    )]
+    pub no_confirm: bool,
+    #[clap(default_value = RECEIPT_LOCATION)]
+    pub receipt: PathBuf,
+    #[clap(long, default_value = PHASE1_RECEIPT_LOCATION)]
+    pub phase1_output: PathBuf,
+    #[clap(long, default_value = PHASE2_RECEIPT_LOCATION)]
+    pub phase2_output: PathBuf,
+    // NOTE(cole-h): an escape hatch in case we somehow run into a case where the "receipt is
+    // valid and we can actually parse it into structs" step does the wrong thing; hidden so
+    // that users aren't tempted to use it themselves, but we can suggest it as a break-glass
+    // measure
+    #[clap(long, hide = true)]
+    pub force_naive_json_method: bool,
+}
+
+#[async_trait::async_trait]
+impl CommandExecute for SplitReceipt {
+    #[tracing::instrument(level = "debug", skip_all)]
+    async fn execute(self) -> eyre::Result<ExitCode> {
+        ensure_root()?;
+
+        let timestamp_millis = SystemTime::now()
+            .duration_since(SystemTime::UNIX_EPOCH)?
+            .as_millis();
+
+        let original_receipt_location = PathBuf::from(RECEIPT_LOCATION);
+        let backed_up_receipt_location = original_receipt_location
+            .with_file_name(format!(".original-receipt.{timestamp_millis}.json"));
+
+        let brief_summary = format!("\n\
+               This will split your existing receipt at {receipt} into two phases (phase 1: {phase1}, phase 2: {phase2}) \
+               for uninstallation purposes, and move the existing receipt to a backup location at {backup_location} afterwards.\n\
+               Phase 1 will clean up everything {except} for the root of the Nix store.\n\
+               Phase 2 will then clean up the root of the Nix store and all its contents.\n\
+               If you are wanting to install with a newer version of nix-installer, \
+               you do not need to run phase 2 of the uninstallation.\n\
+               If you want a clean uninstallation, you should run phase 2 after phase 1.\
```

Comment:
```suggestion
               If you want a clean uninstallation, you should run phase 1, reboot, then run phase 2.\
```

Reply:
I don't want to recommend this because I'm pretty sure as-is (at least on macOS), the Nix Store volume won't be mounted on boot anymore (because we don't mark that step as skipped). I could test if there's a way to keep that step around until after phase 2.